odoo/models.py

```diff
@@ -1699,37 +1723,43 @@ def _get_view(self, view_id=None, view_type='form', **options):
         return arch, view
 
     @api.model
-    def get_view(self, view_id=None, view_type='form', **options):
-        """ get_view([view_id | view_type='form'])
-
-        Get the detailed composition of the requested view like model, view architecture
+    def _get_view_cache_key(self, view_id=None, view_type='form', **options):
+        """ Get the key to use for caching `_get_view_cache`.
+        This method is meant to be overriden by models needing additional keys.
+        :param int view_id: id of the view or None
+        :param str view_type: type of the view to return if view_id is None ('form', 'tree', ...)
+        :param dict options: bool options to return additional features:
+            - bool mobile: true if the web client is currently using the responsive mobile view
+              (to use kanban views instead of list views for x2many fields)
+        :return: a cache key
+        :rtype: tuple
+        """
+        return (view_id, view_type, options.get('mobile'), self.env.lang) + tuple(
+            (key, value) for key, value in self.env.context.items() if key.endswith('_view_ref')
+        )
 
+    @api.model
+    @ormcache('self._get_view_cache_key(view_id, view_type, **options)')
+    def _get_view_cache(self, view_id=None, view_type='form', **options):
+        """ Get the view information ready to be cached
+        The cached view includes the postprocessed view, including inherited views, for all groups.
+        The blocks restricted to groups must therefore be removed after calling this method
+        for users not part of the given groups.
         :param int view_id: id of the view or None
```

Comment:
needs a linebreak before, otherwise rST parsers can't recognize the info fields.